src/multivariate/mvdiscretenonparametric.jl

```diff
@@ -0,0 +1,55 @@
+const MvDiscreteNonParametric{
+    T<:AbstractVector{<:Real},
+    P<:Real,
+    Ts<:AbstractVector{T},
+    Ps<:AbstractVector{P},
+} = GeneralDiscreteNonParametric{Multivariate,T,P,Ts,Ps}
+
+"""
+    MvDiscreteNonParametric(
+        support::AbstractVector,
+        p::AbstractVector{<:Real}=fill(inv(length(support)), length(support)),
+    )
+
+Construct a multivariate discrete nonparametric probability distribution with `support` and corresponding
+probabilities `p`. If the probability vector argument is not passed, then
+equal probability is assigned to each entry in the support.
+
+# Examples
+```julia
+# rows correspond to samples
+x = collect(eachrow(rand(10,2)))
+μ = MvDiscreteNonParametric(x)
+
+# columns correspond to samples
+y = collect(eachcol(rand(7,12)))
+ν = MvDiscreteNonParametric(y)
+```
+"""
+function MvDiscreteNonParametric(
+    support::AbstractArray{<:AbstractVector{<:Real}},
+    p::AbstractVector{<:Real} = fill(inv(length(support)), length(support)),
+)
+    return MvDiscreteNonParametric{eltype(support),eltype(p),typeof(support),typeof(p)}(
+        support,
+        p,
+    )
+end
+
+Base.eltype(::Type{<:MvDiscreteNonParametric{T}}) where T = Base.eltype(T)
+
+function mean(d::MvDiscreteNonParametric)
+    return StatsBase.mean(hcat(d.support...), Weights(d.p, one(eltype(d.p))),dims=2)
+end
+
+function var(d::MvDiscreteNonParametric)
+    x = hcat(support(d)...)
+    p = probs(d)
+    return StatsBase.var(x, Weights(p, one(eltype(p))), 2,corrected = false)
+end
+
+function cov(d::MvDiscreteNonParametric)
+    x = hcat(support(d)...)
+    p = probs(d)
+    return cov(x, Weights(p, one(eltype(p))), 2,corrected = false)
+end
```

Comment:
I requested to add support for vectors of vectors to StatsBase but it might take a bit until it is available. IMO it is fine to have possibly less optimized implementations until then. However, we should avoid splatting and use
```suggestion
function mean(d::MvDiscreteNonParametric)
    return mean(reduce(hcat, d.support), Weights(d.p, one(eltype(d.p))); dims=2)
end

function var(d::MvDiscreteNonParametric)
    x = reduce(hcat, support(d))
    p = probs(d)
    return var(x, Weights(p, one(eltype(p))), 2; corrected = false)
end

function cov(d::MvDiscreteNonParametric)
    x = reduce(hcat, support(d))
    p = probs(d)
    return cov(x, Weights(p, one(eltype(p))), 2; corrected = false)
end
```